tsdb/chunkenc/float_histogram.go

```diff
@@ -358,9 +358,11 @@ func counterResetInAnyFloatBucket(oldBuckets []xorValue, newBuckets []float64, o
 
 		if oldIdx <= newIdx {
 			// Moving ahead old bucket and span by 1 index.
-			if oldInsideSpanIdx == oldSpans[oldSpanSliceIdx].Length-1 {
+			if oldInsideSpanIdx+1 >= oldSpans[oldSpanSliceIdx].Length {
 				// Current span is over.
-				oldSpanSliceIdx++
+				for oldSpanSliceIdx++; oldSpanSliceIdx < len(oldSpans) && oldSpans[oldSpanSliceIdx].Length == 0; oldSpanSliceIdx++ {
+					// Handle patological case of empty span.
```

Comment:
patological → pathological (see also other occurrences below).

As this occurs four times, maybe a short helper `func incrementSpanSliceIdx(idx int, spans []histogram.Span) (newIdx int)` would make sense? (Which the compiler should inline, so no performance penalty.)

Reply:
I'm suggesting the name `nextNonEmptySpanSliceIdx` , verified inline with go version go1.20.4 linux/amd64:
```
./float_histogram.go:363:47: inlining call to nextNonEmptySpanSliceIdx
./float_histogram.go:382:47: inlining call to nextNonEmptySpanSliceIdx
./histogram.go:391:47: inlining call to nextNonEmptySpanSliceIdx
./histogram.go:410:47: inlining call to nextNonEmptySpanSliceIdx
```

Reply:
also added unit test and verified that it fails on main